hal/src/main/native/include/hal/handles/DigitalHandleResource.h

```diff
@@ -38,7 +38,8 @@ class DigitalHandleResource : public HandleBase {
   DigitalHandleResource(const DigitalHandleResource&) = delete;
   DigitalHandleResource& operator=(const DigitalHandleResource&) = delete;
 
-  THandle Allocate(int16_t index, HAL_HandleEnum enumValue, int32_t* status);
+  int32_t Allocate(int16_t index, HAL_HandleEnum enumValue, THandle* handle,
+                   std::shared_ptr<TStruct>* createdOrExisting);
```

Comment:
What's the reason for not having the shared_ptr be the return value?  Doing so would enable RVO and taking `int32_t* status` is more in line with existing functions.

Reply:
No reason. I absolutely could switch it. I'll do that.